kong/runloop/plugins_iterator.lua

```diff
@@ -243,11 +243,6 @@ local function get_next(self)
       local cfg = load_configuration_through_combos(ctx, combos, plugin)
       if cfg then
         plugins[name] = cfg
-
-        if plugin.handler.response then
-          -- there's a "response" handler for this request, make it buffered
-          kong.ctx.core.buffered_proxying = true
-        end
```

Comment:
removing this means any plugin with a `response` phase needs an `access` handler just to turn it on.  for Go plugins that means three extra roundtrips on each request.

Reply:
Can we somehow make it detect it for `go` only plugins? I mean that if people write an abstraction or interface that implements this response function, even when it is empty by default, it will always turn on the buffering. Because it has a lot of issues, we don't want this to be turned on accidentally.

Reply:
I moved it to `response`, as I think we need to have `configuration` to determine whether or not the plugin with response was configured. Only exception is global plugins, but a global plugin with response implemented is a huge no-no.

Reply:
We can probably deprecate `kong.service.request.enable_buffering()` now.